app/src/main/java/com/example/pinterbest/data/repository/AuthRepository.kt

```diff
@@ -0,0 +1,61 @@
+package com.example.pinterbest.data.repository
+
+import com.example.pinterbest.data.exceptions.AlreadyAuthorizedException
+import com.example.pinterbest.data.exceptions.InvalidDataException
+import com.example.pinterbest.data.exceptions.UserExistsException
+import com.example.pinterbest.data.exceptions.UserNotFoundException
+import com.example.pinterbest.data.exceptions.WrongPasswordException
+import java.lang.IllegalStateException
+import okhttp3.ResponseBody
+import retrofit2.Response
+
+class AuthRepository(private val sessionRepository: SessionRepository) {
+
+    fun saveSession(response: Response<ResponseBody>) {
+        val cookie = response.headers().values("Set-Cookie")[0].substringBefore(";")
+        sessionRepository.saveUserData(cookie)
+    }
+
+    fun authProvider(response: Response<ResponseBody>) {
+        return if (response.isSuccessful) {
+            saveSession(response)
+        } else {
+            when (response.code()) {
+                INVALID_DATA -> throwInvalidDataException()
+                WRONG_PASSWORD -> throwWrongPasswordException()
+                ALREADY_AUTHORIZED -> throwAlreadyAuthorizedException()
+                USER_NOT_FOUND -> throwUserNotFoundException()
+                USER_EXISTS -> throwUserExistsException()
+                else -> throw IllegalStateException()
+            }
+        }
+    }
+
+    private fun throwInvalidDataException() {
+        throw InvalidDataException()
+    }
+
+    private fun throwWrongPasswordException() {
+        throw WrongPasswordException()
+    }
+
+    private fun throwAlreadyAuthorizedException() {
+        throw AlreadyAuthorizedException()
+    }
+
+    private fun throwUserNotFoundException() {
+        throw UserNotFoundException()
+    }
+
+    private fun throwUserExistsException() {
+        throw UserExistsException()
+    }
```

Comment:
you don't need this code. Just throw exceptions where needed, without special methods

Reply:
https://detekt.github.io/detekt/style.html#throwscount